#!/usr/bin/env python
PACKAGE = "auv_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp_gain", double_t, 0, "Kp Gain", 0.0, 0, 100)
gen.add("kd_gain", double_t, 0, "Kd Gain", 0.0, 0, 10)
gen.add(
    "navigation_timeout_after_prop_disappear_s",
    double_t,
    0,
    "Navigation Timeout After Prop Disappear",
    0.0,
    0,
    200,
)
gen.add(
    "prop_lost_timeout_s",
    double_t,
    0,
    "Prop Lost Timeout",
    0.0,
    0,
    60,
)
gen.add(
    "search_angular_velocity",
    double_t,
    0,
    "Search Angular Velocity",
    0.0,
    0,
    1.0,
)
gen.add("v_x_desired", double_t, 0, "Desired Forward Velocity", 0.0, 0, 1.0)
gen.add("overall_timeout_s", double_t, 0, "Overall Timeout", 0.0, 0, 1000.0)
gen.add("max_angular_velocity", double_t, 0, "Maximum Angular Velocity", 1.0, 0, 5.0)

exit(gen.generate(PACKAGE, "auv_control", "VisualServoing"))
